src/adagrams.js

```diff
@@ -1,15 +1,167 @@
-export const drawLetters = () => {
-  // Implement this method for wave 1
+// helper
+const countOccurrences = function (array, element) {
+  let count = 0;
+  for (let i = 0; i < array.length; i++) {
+    if (array[i] === element) {
+      count++;
+    }
+  }
+  return count;
 };
+// helper
+const checkForValues = (object, valueOccurernces, letter) =>
+  object[letter] >= valueOccurernces;
 
-export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
-};
+export class Adagrams {
+  constructor() {
+    this.letterhand = 10;
+    this.letterPool = {
+      A: 9,
+      B: 2,
+      C: 2,
+      D: 4,
+      E: 12,
+      F: 2,
+      G: 3,
+      H: 2,
+      I: 9,
+      J: 1,
+      K: 1,
+      L: 4,
+      M: 2,
+      N: 6,
+      O: 8,
+      P: 2,
+      Q: 1,
+      R: 6,
+      S: 4,
+      T: 6,
+      U: 4,
+      V: 2,
+      W: 2,
+      X: 1,
+      Y: 2,
+      Z: 1,
+    };
+    this.scoreValues = {
+      A: 1,
+      B: 3,
+      C: 3,
+      D: 2,
+      E: 1,
+      F: 4,
+      G: 2,
+      H: 4,
+      I: 1,
+      J: 8,
+      K: 5,
+      L: 1,
+      M: 3,
+      N: 1,
+      O: 1,
+      P: 3,
+      Q: 10,
+      R: 1,
+      S: 1,
+      T: 1,
+      U: 1,
+      V: 4,
+      W: 4,
+      X: 8,
+      Y: 4,
+      Z: 10,
+    };
+    this.alphabet = Object.keys(this.scoreValues);
+  }
 
-export const scoreWord = (word) => {
-  // Implement this method for wave 3
-};
+  generateRandomLetter() {
+    return this.alphabet[Math.floor(Math.random() * this.alphabet.length)];
+  }
+  
+  drawLetters = () => {
+    const handList = [];
+    while (handList.length < this.letterhand) {
+      const randomLetter = this.generateRandomLetter();
+      handList.push(randomLetter);
```

Comment:
Instead of pushing and then popping out letters, it may be more efficient to check to see if the letter should even be added before adding it to the hand.